### ISSUE TITLE
read_database errors on read from an empty table

### ISSUE THREAD

**Issue by Sage0614** [pola-rs/polars#14906]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

1. create an empty table with zero records, for example, a table in ms-sql
2. run pl.read_database with a arrow_odbc connection string

3. run
```python
query = f"select count(*) from {table}"
size = pl.read_database(query,conn).get_column("column_0")[0]
```
it will return size = 0, so it is not a connection issue

4. run
```python
query = f"select top 1 * from {table}"
res = pl.read_database(query,conn)
```
it will error out, complaining in /io/database.py/ line 407,258,251, and /convert.py line 614
with ValueError: Must pass schema, or at least one RecordBatch

### Issue description

seems polars is inferring the type from output of the database table record, it assumes the table has at least 1 record. however, if I just want to know the the mapping of database table schema to polars table schema, this method won't work, as there is 0 record in the provided database table.

### Expected behavior

I am expecting I get an empty polars table with the schema matches database table schema, instead of getting an error, 
as "select * from table" is a valid operation on an empty table in database.

### Installed versions

<details>

```
--------Version info---------
Polars:               0.20.13
Index type:           UInt32
Platform:             Linux-5.15.146.1-microsoft-standard-WSL2-x86_64-with-glibc2.35
Python:               3.11.4 (main, Aug 24 2023, 11:18:03) [GCC 11.4.0]

----Optional dependencies----
adbc_driver_manager:  <not installed>
cloudpickle:          <not installed>
connectorx:           <not installed>
deltalake:            <not installed>
fsspec:               2023.6.0
gevent:               <not installed>
hvplot:               <not installed>
matplotlib:           3.7.2
numpy:                1.25.2
openpyxl:             <not installed>
pandas:               2.0.3
pyarrow:              13.0.0
pydantic:             1.10.12
pyiceberg:            <not installed>
pyxlsb:               <not installed>
sqlalchemy:           2.0.20
xlsx2csv:             <not installed>
xlsxwriter:           <not installed>
```

</details>

**Comment by alexander-beedie** [comment]
> seems polars is inferring the type from output of the database table record

We don't actually do that (it would be very unreliable with null values, etc); we actually introspect the resulting cursor description returned, which is typically more robust. I'll look at this in more detail tomorrow, as I can confirm that we don't seem to have a generic problem with empty resultsets (I just quickly tested on a local SQLite db) 🤔

**Comment by Sage0614** [comment]
I see thanks, I encounter this issue with MSSQL and arrow-odbc with ODBC Driver 17 for connection, to be specific.

**Comment by alexander-beedie** [comment]
Got it; was able to reproduce locally. 
The issue is specific to ODBC, but the fix is straightforward 👌

**Comment by alexander-beedie** [comment]
@Sage0614: should be fixed - have a look at the just-released `0.20.15`.

**Comment by Sage0614** [comment]
> @Sage0614: should be fixed - have a look at the just-released `0.20.15`.

Just tried the new version, works correctly for me, thanks!
